crossplane-capi-import/README.md

```diff
@@ -0,0 +1,257 @@
+account and place it into custom resources generated by crossplane.
+
+Each cloud provider will work in similar ways and in order to achieve this,
+we require additional components to be installed into the management cluster.
+
+- crossplane core
+- crossplane-contrib/provider-kubernetes  for the creation of resources inside
+  the cluster
+
+To enable management of resources inside each cloud, the importer will require
+the following additional providers depending on the cloud being managed through
+that management cluster.
+
+| AWS              | Azure                           | GCP                      |
+|------------------|---------------------------------|--------------------------|
+| provider-aws-ec2 | provider-azure-azure            | provider-gcp-compute     |
+| provider-aws-eks | provider-azure-containerservice | provider-gcp-container   |
+
+The full PoC of how this may work for EKS clusters can be found in the
+repository [crossplane-eks-capi-import](https://github.com/giantswarm/crossplane-eks-capi-import/)
+
+Overall the architecture between providers is expected to be similar across
+cloud providers but for the remainder of this RFC, the EKS architecture will be
+used.
+
+![eks-architecture](./crossplane-capi-relationships.png)
+
+### Composition and Definition
+
+The crossplane composition for this importer follows the definition which allows
+for a minimum of the following to be defined
+
+- The region the cluster is built in
+- The cluster name
+- The nodegroup name
+
+This information is then used to look up the cluster details inside the cloud
+provider to be fed to the CAPI resources being generated.
+
+To achieve this lookup, we use crossplane resources for the cloud being searched
+
+| provider-aws-eks                 | provider-azure-containerservice                               | provider-gcp-[container,compute]   |
+|----------------------------------|---------------------------------------------------------------|------------------------------------|
+| Cluster (eks.aws.upbound.io)     | KubernetesCluster (containerservice.azure.upbound.io)         | Cluster (container.gcp.upbound.io) |
+| NodeGroup (eks.aws.upbound.io)   | KubernetesClusterNodePool (containerservice.azure.upbound.io) | NodeGroup (compute.gcp.upbound.io) |
+| ClusterAuth (eks.aws.upbound.io) |                                                               |                                    |
+
+During the generation process, there are certain details that may (excluding the
+use of composition functions) require the editing of specific blocks of the
+composition.
+
+The reason for this is that there is no looping inside the composition and
+details returned from the cloud do not directly correlate to information required
+by CAPI.
+
+The example of this inside EKS is the subnet groups required by the
+`AWSManagedControlPlane` resource which then require hard coded specification
```

Comment:
dumb question - this means even with the XRD, a human has to go in and edit some stuff?

Reply:
At present yes, on our current version of crossplane it is not possible to automatically generate this.

In the next iteration I'm looking at how we can solve this using Composition Functions to remove the remaining requirement on human interaction and make this more automated.

Reply:
cool - with us wanting to productionise this, we'd need to make sure that it's hands off from an operator point of view

Reply:
The functionality to manage tracking subnets from a crossplane composition function now exists: https://github.com/giantswarm/crossplane-fn-generate-subnets/

Reply:
Added a section on how the composition function solves this problem, carrying the link to the PoC and the upstream issue that blocks current further enhancement.